### DISCUSSION TITLE
WindowTabs widget parsing function is parsing the whole widget output instead of individual window name

### DISCUSSION THREAD

**Discussion by jlacours** [qtile/qtile#3741] [Q&A]
Hi, I've been using Qtile for a month and a half now and I love it.

I'm trying to figure out if I'm doing something wrong here.

### Context
I set up the WindowTabs widget to parse window names because they were too long but it doesn't behave the way I expected.

This is my parsing function that removes any extra text and leaves only the name of the window.
``` python
def parse_title(title):
    title_list = ["Firefox"]
    for string in title_list:
        if string in title:
            title = string
        else:
            title = title
    return title
```

This is where I initiate the widget. I'm trying to keep it minimal so nothing interfere with the widget output.
``` python
widget.WindowTabs(
    parse_text = parse_title,
), 
```

This is what I get when a string is detected, as you can see it only shows _Firefox_ even if I have multiples windows open.
![parsing_title_state_1](https://user-images.githubusercontent.com/41354418/182749872-ba76781d-11f0-4efa-bee5-97fdb56c5f62.png)

If Firefox isn't present on the current group, this is what it outputs. (Expected behavior)
![parsing_title_state_2](https://user-images.githubusercontent.com/41354418/182750074-dca0fe9d-466d-47ff-85f5-b47cece74e48.png)

### Conclusion
From my point of you, it looks like the parsing function is parsing the whole output of the widget instead of parsing each individual window name.
Am I doing something wrong or this is a normal behavior?

Thanks in advance for your answers and sorry for my bad english.

**Comment by jlacours** [comment]
The TaskList widget behaves correctly.

**Comment by elParaguayo** [comment]
You are correct. The `parse_text` function parses the entire text of the widget.

**Reply by jlacours** [reply]
Is this the expected behavior or I'm missing something?

**Reply by elParaguayo** [reply]
That is how the widget was written so it is expected.

However, it would seem to make more sense to parse window names individually.

**Reply by jlacours** [reply]
I just started coding, I wish I could help right away. I will try to take a look at it and implement something.
